OpenRA.Mods.Common/HitShapes/Rectangle.cs

```diff
@@ -89,24 +98,43 @@ public WDist DistanceFromEdge(WVec v)
 		public WDist DistanceFromEdge(WPos pos, Actor actor)
 		{
 			var actorPos = actor.CenterPosition;
+			var orientation = new WRot(actor.Orientation.Roll, actor.Orientation.Pitch,
+				new WAngle(actor.Orientation.Yaw.Angle + (RotateToIsometry ? 128 : 0)));
+
+			var targetablePositions = actor.TraitsImplementing<ITargetablePositions>();
+			if (ApplyToAllTargetablePositions && targetablePositions.Any())
+			{
+				var positions = targetablePositions.SelectMany(tp => tp.TargetablePositions(actor));
+				actorPos = positions.PositionClosestTo(pos);
+			}
 
 			if (pos.Z > actorPos.Z + VerticalTopOffset)
-				return DistanceFromEdge((pos - (actorPos + new WVec(0, 0, VerticalTopOffset))).Rotate(-actor.Orientation));
+				return DistanceFromEdge((pos - (actorPos + new WVec(0, 0, VerticalTopOffset))).Rotate(-orientation));
 
 			if (pos.Z < actorPos.Z + VerticalBottomOffset)
-				return DistanceFromEdge((pos - (actorPos + new WVec(0, 0, VerticalBottomOffset))).Rotate(-actor.Orientation));
+				return DistanceFromEdge((pos - (actorPos + new WVec(0, 0, VerticalBottomOffset))).Rotate(-orientation));
 
-			return DistanceFromEdge((pos - new WPos(actorPos.X, actorPos.Y, pos.Z)).Rotate(-actor.Orientation));
+			return DistanceFromEdge((pos - new WPos(actorPos.X, actorPos.Y, pos.Z)).Rotate(-orientation));
 		}
 
 		public void DrawCombatOverlay(WorldRenderer wr, RgbaColorRenderer wcr, Actor actor)
 		{
 			var actorPos = actor.CenterPosition;
+			var orientation = new WRot(actor.Orientation.Roll, actor.Orientation.Pitch,
+				new WAngle(actor.Orientation.Yaw.Angle + (RotateToIsometry ? 128 : 0)));
 
-			var vertsTop = combatOverlayVertsTop.Select(v => wr.ScreenPosition(actorPos + v.Rotate(actor.Orientation)));
-			var vertsBottom = combatOverlayVertsBottom.Select(v => wr.ScreenPosition(actorPos + v.Rotate(actor.Orientation)));
-			wcr.DrawPolygon(vertsTop.ToArray(), 1, Color.Yellow);
-			wcr.DrawPolygon(vertsBottom.ToArray(), 1, Color.Yellow);
+			var targetablePositions = actor.TraitsImplementing<ITargetablePositions>();
+			if (ApplyToAllTargetablePositions && targetablePositions.Any())
```

Comment:
Oh, I see. You effectively only draw anything when `ApplyToAllTargetablePositions` is set, and you didn't set it for gates.